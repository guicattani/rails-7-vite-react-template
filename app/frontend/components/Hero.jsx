import { useState, useEffect } from "react";

export default function Hero() {
  const [greeting, setGreeting] = useState("");

  useEffect(() => {
    fetch("/api/v1/home")
      .then((response) => {
        return response.json();
      })
      .then((data) => {
        setGreeting(data);
      });
  }, []);

  return (
    <>
      <h1 role="heading" className="hero">
        <span className="vite smooth">Vite</span>{" "}
        <span className="rails smooth">Rails</span>
      </h1>

      <h2> {greeting && <p>{greeting}</p>}</h2>
    </>
  );
}
