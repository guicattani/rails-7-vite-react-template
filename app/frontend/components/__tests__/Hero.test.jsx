import React from "react";
import { act } from "react-dom/test-utils";
import { screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom/client";
import fetch from "cross-fetch";
import crossFetch from "cross-fetch";
global.fetch = fetch;

import Hero from "../Hero";

jest.mock("cross-fetch", () => {
  //Mock the default export
  return {
    __esModule: true,
    default: jest.fn(),
  };
});

let container;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);

  crossFetch.mockResolvedValue({
    status: 200,
    json: () => "greetings",
  });
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe("Hero", () => {
  it("should show text Vite Rails", async () => {
    act(() => {
      ReactDOM.createRoot(container).render(<Hero />);
    });

    await waitFor(() => {
      expect(screen.getByText("Vite")).toBeInTheDocument();
      expect(screen.getByText("Rails")).toBeInTheDocument();
    });
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<Hero />);
    expect(tree).toMatchSnapshot();
  });
});
