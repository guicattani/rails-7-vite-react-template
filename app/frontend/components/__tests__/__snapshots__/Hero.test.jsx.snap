// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should match the snapshot 1`] = `
[
  <h1
    className="hero"
    role="heading"
  >
    <span
      className="vite smooth"
    >
      Vite
    </span>
     
    <span
      className="rails smooth"
    >
      Rails
    </span>
  </h1>,
  <h2>
     
  </h2>,
]
`;
